hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HoodieArrayWritableAvroUtils.java

```diff
@@ -37,12 +38,17 @@ public class HoodieArrayWritableAvroUtils {
   private static final Cache<String, ObjectInspectorCache>
       OBJECT_INSPECTOR_TABLE_CACHE = Caffeine.newBuilder().maximumSize(1000).build();
 
-  public static ObjectInspectorCache getCacheForTable(String table, Schema tableSchema, JobConf jobConf) {
-    ObjectInspectorCache cache = OBJECT_INSPECTOR_TABLE_CACHE.getIfPresent(table);
-    if (cache == null) {
-      cache = new ObjectInspectorCache(tableSchema, jobConf);
-    }
-    return cache;
+  public static ObjectInspectorCache getCacheForTable(String table) {
+    return OBJECT_INSPECTOR_TABLE_CACHE.getIfPresent(table);
+  }
+
+  public static ObjectInspectorCache initCacheForTable(String table, Schema tableSchema, JobConf jobConf) {
+    return OBJECT_INSPECTOR_TABLE_CACHE.get(table, t -> new ObjectInspectorCache(tableSchema, jobConf));
+  }
+
+  @VisibleForTesting
+  public static void resetCache() {
```

Comment:
needed to add this because now that we cache things, there were conflicts between tests